### PR TITLE
Updated Spring Boot to version 3.3.5

### PR DESCRIPTION
**Pull request by gr4cza**
- removed unnecessary restriction from `tomcat-embed-core`

Pull Request type
----
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] WHOSUSING.md
- [ ] Other (please describe):

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

Changes in this PR
----

Updated Spring Boot to version 3.3.5, as this version does not include the vulnerability that previously required restricting the org.apache.tomcat.embed:tomcat-embed-core dependency.

### CODE REVIEW

**Comment by gr4cza** [comment]
hi @v1r3n!
can you take a look on this one?
the 
```
implementation('org.apache.tomcat.embed:tomcat-embed-core') {
    version {
        strictly '10.1.25'
    }
}
```
part prevents the further spring boot updates (without explicit exlusions).

thank you!

File: dependencies.gradle [hunk 1]
```diff
@@ -28,12 +28,12 @@ ext {
...
     revEurekaClient = '2.0.2'
     revGroovy = '4.0.9'
-    revGrpc = '1.57.2'
+    revGrpc = '1.66.0'
     revGuava = '33.2.1-jre'
     revHamcrestAllMatchers = '1.8'
     revHealth = '1.1.4'
     revPostgres = '42.7.2'
-    revProtoBuf = '3.21.12'
+    revProtoBuf = '3.25.3'
```
Comment (c4lm) [inline dependencies.gradle:36]: could you try 3.25.5? I know upgrading anything related to protobuf and grpc can be quite a PITA due to what they pull vs what other libraries relying on them pull, but if 3.25.5 works, we should use that - it is vulnerability-free at the moment

Comment (gr4cza) [reply dependencies.gradle]: let see :D

Comment (gr4cza) [reply dependencies.gradle]: let's see 😄

Comment (gr4cza) [reply dependencies.gradle]: it seems fine to me 👍

**Review by c4lm — APPROVED** [review]

**Pull request merged**